bson/src/main/org/bson/internal/ProvidersCodecRegistry.java

```diff
@@ -14,20 +14,22 @@
  * limitations under the License.
  */
 
-package org.bson.codecs.configuration;
+package org.bson.internal;
 
 import org.bson.codecs.Codec;
+import org.bson.codecs.configuration.CodecProvider;
+import org.bson.codecs.configuration.CodecRegistry;
 
 import java.util.ArrayList;
 import java.util.List;
 
 import static org.bson.assertions.Assertions.isTrueArgument;
 
-final class ProvidersCodecRegistry implements CodecRegistry, CodecProvider {
+public final class ProvidersCodecRegistry implements CodecRegistry, CodecProvider, CycleDetectingCodecRegistry {
```

Comment:
Are there any tests proving that this is Cycle Detecting?

Reply:
No, will add.

Reply:
Done